gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+    url = 'http://172.24.4.5:8888'
+    url_for_add_inte = "%s/add_rule" % url
+    url_for_del_inte = "%s/delete_rule" % url
+    url_for_add_src_route = "%s/add-source-route" % url
+    url_for_del_src_route = "%s/delete-source-route" % url
+    url_for_config_fw = "%s/configure-firewall-rule" % url
+    url_for_update_fw = "%s/update-firewall-rule" % url
+    url_for_delete_fw = "%s/delete-firewall-rule" % url
+    data = '{"stitching_mac": "00:0a:95:9d:68:25", "provider_mac": "00:0a:95:9d:68:16"}'
+    data_for_interface = '{"stitching_mac": "00:0a:95:9d:68:25", "provider_mac": "00:0a:95:9d:68:16"}'
+    data_for_add_src_route = '[{"source_cidr": "1.2.3.4/24", "gateway_ip": "1.2.3.4"}]'
+    data_for_del_src_route = '[{"source_cidr": "1.2.3.4/24"}]'
+    timeout = 30
+
+    def fake_request_data_generic_bulk(self):
+        request_data = {
+            "info": {
+                "version": 1
+            },
+            "config": [{
+                "resource": "interfaces",
+                "kwargs": {
+                    "stitching_mac": None,
+                    "context": {
+                        "domain": None},
+                    "mgmt_ip": "120.0.0.15",
+                    "service_vendor": "vyos",
+                    "request_info": {
+                        "network_function_id": "5084bb63-459f-4edd-a090-97168764e632",
+                        "network_function_device_id": "9b7f6b15-3e67-4f26-8463-b4efb36cbb08",
+                        "network_function_instance_id": "940dcdf3-77c8-4119-9f95-ec1e16a50fa8"
+                    },
+                    "provider_mac": "fa:16:3e:0f:0f:06",
+                    "provider_interface_position": 2,
+                    "stitching_cidr": None,
+                    "provider_ip": "11.0.0.1",
+                    "stitching_interface_position": 3,
+                    "provider_cidr": "11.0.0.0/24",
+                    "stitching_ip": None
+                }
+            }, {
+                "resource": "routes",
+                "kwargs": {
+                    "mgmt_ip": "120.0.0.15",
+                    "service_vendor": "vyos",
+                    "request_info": {
+                        "network_function_id": "5084bb63-459f-4edd-a090-97168764e632",
+                        "network_function_device_id": "9b7f6b15-3e67-4f26-8463-b4efb36cbb08",
+                        "network_function_instance_id": "940dcdf3-77c8-4119-9f95-ec1e16a50fa8"
+                    },
+                    "provider_interface_position": 2,
+                    "gateway_ip": None,
+                    "destination_cidr": None,
+                    "context": {
+                        "domain": None},
+                    "source_cidrs": ["11.0.0.0/24"]
+                }
+            }]
+        }
+        return request_data
+    
+    def fake_request_data_generic_single(self):
+        request_data = self.fake_request_data_generic_bulk()
+        request_data['config'].pop()
```

Comment:
This logic will fail if you have three request data blobs in the single request data.